server/server.go

```diff
@@ -605,6 +606,17 @@ func (s *Server) v1DiagnosticsGet(w http.ResponseWriter, r *http.Request) {
 	writer.JSON(w, 200, resp, pretty)
 }
 
+func (s *Server) v1VersionGet(w http.ResponseWriter, r *http.Request) {
```

Comment:
It would be better to return the version information as a JSON object. For example:

```
{
  "result": {
    "version": "0.9.3-dev",
    "build_commit": "1234abcd",
    "build_timestamp": "2000-01-01T00:00:00Z",
    "build_hostname": "buildbox"
  }
}
```

Note, the version object should be returned under the "result" key to be consistent with the rest of the Data API.

Reply:
I see a problem with the version, build_commit, etc. At least in my Mac they are not set during build time as they should. I can only see them if I actually run OPA.

I can see in the Makefile that they should be set, but they are not.

LDFLAGS := "-X github.com/open-policy-agent/opa/version.Version=$(VERSION) \
	-X github.com/open-policy-agent/opa/version.Vcs=$(BUILD_COMMIT) \
	-X github.com/open-policy-agent/opa/version.Timestamp=$(BUILD_TIMESTAMP) \
	-X github.com/open-policy-agent/opa/version.Hostname=$(BUILD_HOSTNAME)"

But after make the file version.go does not have any data. Actually this is true even for *existing tests*.

Reply:
As an example, if I run test TestIndexGet(), which is an existing test, I can see that version, timestamp, etc are all empty.

Reply:
If you're building OPA without the makefile then you need to specify the LDFLAGS yourself at build-time. The makefile is not specifying them when running tests so that's why they're not set in the tests.

The version.go file will not change during the build, the -X flags just set variables at link time.

In any case, it's fine to leave the test as is for now. Let's just update the response representation.

Reply:
I am building with the Makefile. I go to OPA dir and issue <make>, it goes through everything.

So, you are saying there is no way to check the actual version, etc during tests and we can leave as it is?

Reply:
It looks like `go test` supports ldflags now (I'm not sure it always did...) Once I merge the other PR you should be able to refer to version.Version in your test.

Reply:
Sounds good, let me make the change

Reply:
Updated, should be good now. We probably need a different g\Github issue if we want the data in version.go to match those in indexGet, VersionGet and possibly others.